File: python/logout.py
#!/usr/bin/python3

"""
    logout.py

    MediaWiki API Demos
    Demo of `Logout` module: Log out and clear session data.

    MIT License
"""

import requests

S = requests.Session()

URL = "https://en.wikipedia.org/w/api.php"

# Retrieve login token first
PARAMS_0 = {
    'action':"query",
    'meta':"tokens",
    'type':"login",
    'format':"json"
}

R = S.get(url=URL, params=PARAMS_0)
DATA = R.json()

LOGIN_TOKEN = DATA['query']['tokens']['logintoken']

print(LOGIN_TOKEN)

# Send a POST request to login. Using the main account for login is not
# supported. Obtain credentials via Special:BotPasswords
# (https://www.mediawiki.org/wiki/Special:BotPasswords) for lgname & lgpassword

PARAMS_1 = {
    'action':"login",
    'lgname':"your_bot_username",
    'lgpassword':"your_bot_password",
    'lgtoken':LOGIN_TOKEN,
    'format':"json"
}

R = S.post(URL, data=PARAMS_1)
DATA = R.json()

print(DATA)

# GET request to fetch CSRF token
PARAMS_2 = {
    "action":"query",
    "meta":"tokens",
    "format":"json"
}

R = S.get(url=URL, params=PARAMS_2)
DATA = R.json()

CSRF_TOKEN = DATA['query']['tokens']['csrftoken']

print( CSRF_TOKEN )

# Send a POST request to logout.
PARAMS_3 = {
    "action": "logout",
    "token": CSRF_TOKEN,
    "format": "json"
}

R = S.post(URL, data=PARAMS_3)
DATA = R.json()

# Check weither the response is null or not
if DATA == {}:
    print("You have successfully logout.")

File: javascript/logout.js
/*
    logout.js

    MediaWiki API Demos
    Demo of `Logout` module: Log out and clear session data.

    MIT License
*/

var request = require('request').defaults({jar: true}),
    url = "https://en.wikipedia.org/w/api.php";

// Retrieve login token first
function getLoginToken() {
    var params_0 = {
        action: "query",
        meta: "tokens",
        type: "login",
        format: "json"
    };

    var query = url + "?";

    Object.keys(params_0).forEach(function (key) {
        query += "&" + key + "=" + params_0[key];
    });

    request.get(query, function (error, res, body) {
        if (error) {
            return;
        }
        var data = JSON.parse(body);
        loginRequest(data.query.tokens.logintoken);
    });
}

// POST Request to login
// Use of main account for login is not
// supported. Obtain credentials via Special:BotPasswords
// (https://www.mediawiki.org/wiki/Special:BotPasswords) for lgname & lgpassword
function loginRequest(login_token) {
    var params_1 = {
        action: "login",
        lgname: "your_bot_username",
        lgpassword: "your_bot_password",
        lgtoken: login_token,
        format: "json"
    };

    request.post({
        url: url,
        form: params_1,
    }, function (error, res, body) {
        if (error) {
            return;
        }
        getCsrfToken();
    });
}

// GET request to fetch CSRF token
function getCsrfToken() {
    var params_2 = {
        action: "query",
        meta: "tokens",
        format: "json"
    };

    var query = url + "?";

    Object.keys(params_2).forEach(function (key) {
        query += "&" + key + "=" + params_2[key];
    });

    request.get(query, function(error, res, body) {
        if (error) {
            return;
        }
        var data = JSON.parse(body);
        logoutRequest(data.query.tokens.csrftoken);
    });
}

// Send a POST request to logout
function logoutRequest(csrf_token) {
    var params_3 = {
        action: "logout",
        token: csrf_token,
        format: "json"
    };

    request.post({
        url: url,
        form: params_3,
    }, function (error, res, body) {
        if (error) {
            return;
        }

        // Check weither the response is null or not
        if ( body == '{}' ){
            console.log("You have successfully logout.");
        }
    });
}

// Start From Step 1
getLoginToken();